content/chapters/software-stack/lab/content/basic-syscall.md

```diff
@@ -3,6 +3,27 @@
 To get a better grasp on how the software stack works, let's do a bottom-up approach:
 we build and run different programs, that start of by using the system call API (the lowest layer in the software stack) and progressively use higher layers.
 
+### System Calls Explained
+
+A system call, or syscall for short, is a method used by applications to communicate with the operating system's kernel.
+
+The need for syscalls is tied to the modern operating systems model of conceptually separating into kernel space and user space.
+
+The kernel space manages the hardware resources such as CPU, I/O devices, disk or memory.
+Moreover, the kernel also provides an interface for the user space applications to interact with the hardware.
+
+The user space is where you are running your applications and processes.
+From the user space, we cannot directly access the hardware or perform privileged operations.
+You need to use syscalls to perform privileged operations such as accessing the hardware.
+
+Below, you can see some examples of system calls and what resource they request from the kernel:
+
+- `brk()` is used to allocate memory
+- `open()` is used to access the file system and open a specific file
+- `write()` is used to access the file system and modify the contents of a specific file
+
+<img src="../media/syscall-explained.svg" alt="System Calls Explained" width="40%" />
```

Comment:
Don't use inline HTML because it messes up the rendered website (https://github.com/open-education-hub/operating-systems/pull/108#issue-1538121478). Make the image smaller instead.

Reply:
Resizing is optional, from what I have tested Docusaurus resizes images if the size is below a threshold. We should only focus on the content right now.

Reply:
Done